### PR TITLE
Try/variation icons editor

### PR DESCRIPTION
**Pull request by gaambo**
Tries solution no 3 for #21.
The icon and label for social sharing link variations (=services) will be get from all registered block variations (the data store) and not from the hardcoded variations of the plugin. 

**Testing instructions:**
1. Register a custom block variation for the `outermost/social-sharing-link` block in JavaScript (editor): 
```js
import { link as linkIcon } from "@wordpress/icons";
registerBlockVariation("outermost/social-sharing-link", {
    name: "link",
    isDefault: true,
    title: __("Link"),
    icon: linkIcon,
    attributes: {
      service: "link",
    },
    isActive: (blockAttributes, variationAttributes) => blockAttributes.service === "link",
  });
```
2. Add a social sharing block in the editor and add this variation as a child. 
3. The block displays the correct icon and label (renderd in the button!)

**Important:** Does only work with icons set to SVG/react components. If you pass a string (like the name of a Dashicon) the string directly will be rendered.

**TODO:**
- [ ] Maybe check that icon is a component and not a string
- [ ] Map Twitter to X

**Important 2**
This is based on top of the [add/local-dev-environment branch](https://github.com/gaambo/social-sharing-block/tree/add/local-dev-environment) of my fork. I can cherry pick only the one commmit required for this issue (e074114a0abd0a3b92f3fa22914cd63878b62886) or (if PR #20 gets merged) rebase this branch on the upstream main.

### CODE REVIEW

File: src/social-sharing-link/social-list.js [hunk 1]
```diff
@@ -6,55 +6,70 @@ import { find } from 'lodash';
...
-	// Transition Twitter to X
-	if ( 'twitter' === name ) {
-		return __( 'X', 'social-sharing-block' );
-	}
-
-	const variation = find( variations, { name } );
-
-	if ( ! variation ) {
-		return __( 'Mail', 'social-sharing-block' );
+export const useVariationLabel = (attributes) => {
```
Comment (philcable) [inline src/social-sharing-link/social-list.js]: These two functions share a lot of logic - would it make any sense to consolidate them into one that returns an object with `icon` and `label` properties?

**Review by philcable — COMMENTED** [review]
Hi @gaambo, thank you very much for diving into this! It was completely unexpected and much appreciated.

I left one thought on the approach - feel free to take it or leave it 🙂!

File: src/social-sharing-link/social-list.js (hunk 1 above)
Comment (gaambo) [reply src/social-sharing-link/social-list.js]: Thanks, that's good feedback :) Did that in b31b5ae0ce084e9fa32eaa07be14721e22749a9d

**Comment by ndiego** [comment]
Closing in favor of https://github.com/ndiego/social-sharing-block/pull/23. Thanks so much for outlining the solution here!

**Pull request closed without merging**